client.go

```diff
@@ -79,24 +93,34 @@ func (client *Client) Close() (err error) {
 	// 		err = fmt.Errorf("%v", e)
 	// 	}
 	// }()
+	C.Clear(client.api)
 	C.Free(client.api)
 	return err
 }
 
 // SetImage sets path to image file to be processed OCR.
 func (client *Client) SetImage(imagepath string) *Client {
+	if client.pixImage != nil {
+		C.DestroyPixImage(client.pixImage)
+		client.pixImage = nil
+	}
 	client.ImagePath = imagepath
 	return client
 }
```

Comment:
seems I should change the interface of some methods to return errors explicitly, such like

```go
func (client *Client) SetImage(imagepath string) error
```

@otiai10 I will do it after it gets merged

Reply:
I think that would be beneficial since we are using Leptonica on the c side and not checking any errors c side. SetImage should check that the imagepath is not blank and that it is a valid path on the os. I added these to prepare function.